### ISSUE TITLE
Build issues on macOS/cmake

### ISSUE THREAD

**Issue by kvark** [DiligentGraphics/DiligentEngine#78]
Doing a fresh checkout, configuring Cmake from a new `build` folder and calling `make` gets me multiple python2 errors:
```
Validating DiligentCore module's source code formatting...
ERROR:root:code for hash md5 was not found.
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/hashlib.py", line 147, in <module>
    globals()[__func_name] = __get_hash(__func_name)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/hashlib.py", line 97, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type md5
```
Which are eventually followed by:
```
[ 65%] Building C object DiligentSamples/Samples/Atmosphere/CMakeFiles/Atmosphere.dir/__/__/__/DiligentTools/NativeApp/Apple/Source/main.m.o
error: invalid argument '-std=gnu11' not allowed with 'Objective-C++'
make[2]: *** [DiligentSamples/Samples/Atmosphere/CMakeFiles/Atmosphere.dir/__/__/__/DiligentTools/NativeApp/Apple/Source/main.m.o] Error 1
make[1]: *** [DiligentSamples/Samples/Atmosphere/CMakeFiles/Atmosphere.dir/all] Error 2
```

- OS: MacOS Catalina 10.15
- Build tools: Cmake 3.17.0

**Comment by TheMostDiligent** [comment]
Can you try to update python to 3.0?

**Comment by kvark** [comment]
Brew does have Python 3.7 installed. Not sure what else is needed.
Also `error: invalid argument '-std=gnu11' not allowed with 'Objective-C++'` is not related to Python.

**Comment by TheMostDiligent** [comment]
It appears it is not finding python 3.0 as [it should](https://travis-ci.org/github/DiligentGraphics/DiligentCore/jobs/670855170#L145).

**Comment by kvark** [comment]
I was able to resolve the python errors by following https://opensource.com/article/19/5/python-3-default-mac (perhaps, write something to your build instructions, or detect if it's not python2?).

The error `error: invalid argument '-std=gnu11' not allowed with 'Objective-C++'` is still blocking the build

**Comment by TheMostDiligent** [comment]
Just tried to do clean build and I can't reproduce this on XCode Version 11.1 (11A1027), compiler version 11.0.0.11000033.
The build also completes OK on CI server with the [same compiler](https://travis-ci.org/github/DiligentGraphics/DiligentCore/jobs/670855170#L123).

Which XCode/compiler do you use?
Also, did you try to build the project from XCode rather than command line?

As to `-std=gnu11`, I see the flag in compiler flags for every project. I did not figure out where it comes from. I also don't know why it applies to objective-c files.

**Comment by kvark** [comment]
When I switched `cmake` to XCode, like your documentation suggests, it built fine.

**Comment by kvark** [comment]
After a few fixes, got it working on [gfx-portability](https://github.com/gfx-rs/portability) on macOS/Metal (again): ![screenshot](https://user-images.githubusercontent.com/107301/78851397-b045eb80-79e7-11ea-9b3e-785c97196cec.png)

Please reach out if you are interested in benchmarking. Given some instructions, I could run it against MoltenVK on a few machines to see if we can get you moare speed ;)

**Comment by TheMostDiligent** [comment]
This is what it should actually look like:
![image](https://user-images.githubusercontent.com/61806567/78851639-25f18d80-79cf-11ea-86ee-bbd33fbd3684.png)
Did you try to run it with MoltenVK?

**Comment by kvark** [comment]
hah, good point! So the sky as well as atmospheric effects are broken.
Will look more into it, too early to bench anything :)

**Comment by kvark** [comment]
@TheMostDiligent running with MoltenVK from the SDK version 1.2.131.2, I'm hitting the following assertion at submit:
> validateComputeFunctionArguments:857: failed assertion `Compute Function(main0): missing sampler binding at index 0 for _92Smplr[0].'

What are you running with?

**Comment by kvark** [comment]
Another problem I have on gfx-portability is that the driver hangs (Intel Iris 550) quite soon after launch of Atmosphere. Could be related to the missing sky. Perhaps, also to these validation warnings?
> Diligent Engine: ERROR: Vulkan debug message (validation): UNASSIGNED-features-limits-maxComputeWorkGroupSize
                 VkShaderModule 0x4720000000472[RefineSampleLocationsCS] local_size_x (128) exceeds device limit maxComputeWorkGroupSize[0] (64).
                 Object[0] (shader module): Handle 0x4720000000472, Name: 'RefineSampleLocationsCS'

**Comment by TheMostDiligent** [comment]
I currently have 1.1.130.0. Looks like I will need to test it and open an issue vs MoltenVK

**Comment by TheMostDiligent** [comment]
The thread group size may be causing the issue, hard to tell. I don't have this problem on Intel Iris Graphics 6100. TBH, 64 is too low of a bar even for Intel GPUs...

**Comment by kvark** [comment]
I'll double check my exposed limits. Diligent should still make sure it doesn't submit invalid work.

**Comment by TheMostDiligent** [comment]
I agree. However, with this particular limit it is very much like checking if the device supports 2k x 2k texture resolution. Any Vulkan-capable device should be able to do better than this.

**Comment by kvark** [comment]
Yes. I updated the limits and it still hangs the driver :/

**Comment by TheMostDiligent** [comment]
Did you try to run it with 1.1.130.0 MoltenVK?

**Comment by TheMostDiligent** [comment]
So I tested the sample with the latest LunarG SDK and there is indeed an error in 1.1.131.1 and 1.1.131.2 SDKs. However, they just released 1.2.135 SDK and the sample runs fine again.

**Comment by TheMostDiligent** [comment]
All samples and Tutorials run fine with MoltenVK 1.2.135  on MacOS Catalina 10.15.3 with Intel Iris Graphics 6100.
The only exception is compute shader tutorial, but it has never worked on Mac due to missing support of atomics.
Note that they changed the paths to loader files from etc to share (I updated [the instructions](https://github.com/DiligentGraphics/DiligentEngine#configuring-vulkan-build-environment))
